h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java

```diff
@@ -153,4 +189,23 @@ public AutoMLBuildModelsV99() {
   @API(help="The AutoML Job key", direction=API.Direction.OUTPUT)
   public JobV3 job;
 
+
+  @Override
+  public AutoMLBuildSpecV99 fillFromBody(String body) {
+    AutoMLBuildSpecV99 schema = super.fillFromBody(body); //default JSON filling
+
+    // TODO: need to understand why we set stopping tolerance to AUTO iff stopping_criteria is provided without stopping_tolerance
```

Comment:
@seb-h2o Have you resolved this TODO?  Judging just by what is written in TODO it looks logical that if we provide `stopping_criteria` and no `stopping_tolerance` has been specified then we should assign it to automatically selected value.

Reply:
my interrogation was that not in the `if` but in the `iff`, meaning the `only if` part.
if no `stopping_criteria` is provided at all, then `stopping_tolerance` is not set to `AUTO` but to a default value of `0.001`, which makes those default value difficult to.
To sum up:
```
stopping_criteria = null    ->  stopping_tolerance = 0.001
stopping_criteria = {}       ->  stopping_tolerance = AUTO  // = -1
stopping_criteria = {max_runtime_secs = 300}       ->  stopping_tolerance = AUTO  // = -1
```
You have to agree that it is very confusing, and I'm considering it as a bug indeed.
@ledell are you aware of this?

Reply:
Note that this is not an issue with Py/R/Flow clients as stopping_criteria can't be null there, therefore we get `stopping_tolerance = AUTO` if not set.
But for those using Java or REST API directly, this is inconsistent.
For Sparkling Water, there may be the same inconsistency.

Reply:
@seb-h2o I see now, for me it had equally "not provided" meaning for `null` and `{}` and any other things that denote non-existence/emptiness. Thank you for clarification.